### ISSUE TITLE
Adding Route53 records with nested double quotes escaped, causes Terraform plan to always show the resource as modified.

### ISSUE THREAD

**Issue by c4milo** [hashicorp/terraform#8423]
### Background

Route53 does not accept record values longer than 255 characters. This makes adding [DKIM](https://en.wikipedia.org/wiki/DomainKeys_Identified_Mail) TXT records a little bit tricky. However, it seems there is a way to do it by splitting the value using `[double quoute][space][double quote]` as explained in https://forums.aws.amazon.com/thread.jspa?threadID=93940

I was able to create the DKIM record by escaping the nested double qoutes, but Terraform plan keeps showing the resource as modified every time. 
### Terraform Version

v0.7.2-dev
### Affected Resource(s)
- aws_route53_record
### Terraform Configuration Files

``` hcl
resource "aws_route53_record" "google_dkim" {
    zone_id = "${aws_route53_zone.primary.zone_id}"
    name = "google._domainkey"
    type = "TXT"
    ttl = "300"
    records = [
        "v=DKIM1; k=rsa; p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAiajKNMp\" \"/A12roF4p3MBm9QxQu6GDsBlWUWFx8EaS8TCo3Qe8Cj0kTag1JMjzCC1s6oM0a43JhO6mp6z/\" \"MzVf01GWGxxvrtv0lJLHBjWSQqDTQwAZAsa/eKUrsDTbTHYDgXwKeYB2W192oNOTf1oy/\" \"CHvvDsPqnavOYBg8/y18SKjfrc768n4t41ZZcs6GRX4erMlspsUkS6cmSpsyu/\" \"OpMzM1qq05PduKs26VfD2r+mQYdB6D1QOeYodhhNNFxRxMZeepbh/\" \"SRiV3AbLlawTYJbpKWuFfBGT0j1ieNKHVyNdssYf4DWveP2pjNs0MePx/hVASkCM3JkXH8MfrtF/nwIDAQAB"
    ]
}
```
### Expected Behavior

Terraform plan should not show the resource as modified as the record is already created in Route53.
### Actual Behavior

Terraform plan shows the resource as modified despite the record value being the same.
### Steps to Reproduce
1. Just try creating a Route53 TXT record with a value longer than 255 characters, split as explained above.

**Comment by cakoose** [comment]
In Route53, you sometimes have to quote TXT records.  The Terraform Route53 plugin tries to handle the quoting for you automatically, so you don't have to put the quotes in your config.  Unfortunately, the code to add/remove quotes is hacky and doesn't handle all cases, like the case of multiple quoted strings.

In particular, the [quote-removal code](https://github.com/hashicorp/terraform/blob/05c83a3ca19076c79fb93c0ca33bf332236c2a2a/builtin/providers/aws/structure.go#L812)) will remove the first two quote characters in the string.  So if you have:

    "abc xyz" "blah blah"

It gets transformed to:

    abc xyz "blah blah"

I'm not sure of the perfect way to fix this, but one option is to change the quote-removal code to remove quotes from the front and back of the string.

**Comment by cakoose** [comment]
I submitted a pull request to make the situation slightly better (just enough for this use case to work).

However, a more complete (but also more invasive solution) might be:
1. When pulling the config from AWS, parse the entry properly to unquote+unescape everything into single raw string.
2. When pushing the config to AWS, automatically add quotes and split on 255-char boundaries.

This would mean your config won't always round-trip the first time (since you might have split up your string differently in the UI) but after one `terraform apply` it should stay stable.

**Comment by radeksimko** [comment]
Closed via #14170 
Thanks @cakoose
